src/main/java/uk/gov/pay/api/json/RequestJsonParser.java

```diff
@@ -117,7 +118,15 @@ static CreateCardPaymentRequest parsePaymentRequest(JsonNode paymentRequest) {
 
     static CreateAgreementRequest parseAgreementRequest(JsonNode agreementRequest) {
         var builder = CreateAgreementRequestBuilder.builder()
-                .reference(validateAndGetReference(agreementRequest));
+                .reference(validateAndGetReference(agreementRequest))
+                .description(validateAndGetDescription(agreementRequest));
+
+        if (agreementRequest.has(USER_IDENTIFIER_FIELD)) {
+            String userIdentifier = validateSkipNullValueAndGetString(agreementRequest.get(USER_IDENTIFIER_FIELD),
+                    aPaymentError(USER_IDENTIFIER_FIELD, CREATE_PAYMENT_VALIDATION_ERROR, "Field must be a string"));
```

Comment:
Not too sure about reusing `CREATE_PAYMENT_VALIDATION_ERROR` but we do such things elsewhere and I promised I’d create a story to review that, so I just did (PP-9498). So this can stay as it is for now.